### ISSUE TITLE
The accordion toggles for the filters present their role as "link" instead of button, and do not specify their state.

### ISSUE THREAD

**Issue by christinach** [pulibrary/orangelight#2821]
Assign role="button" and aria-expanded="[true/false]", and make sure the element reacts to spacebar as well as enter.

https://www.w3.org/WAI/WCAG21/quickref/#name-role-value

https://github.com/projectblacklight/blacklight/issues/1905

**Comment by christinach** [comment]
from https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role
- [ ] add role="button" (so that we dont need to change the facet to a `<button>` or `<input type="button">`)
- [ ] make the element focusable
- [ ] define event handlers for click and keydown events. (This includes handling the Enter and Space keypresses in order to process all forms of user input. See [the official WAI-ARIA example code](https://www.w3.org/TR/wai-aria-practices/examples/button/button.html).)
